include/ygm/container/oversized_bag.hpp

```diff
@@ -0,0 +1,335 @@
+      std::convertible_to<typename STLContainer::value_type, Item>
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+
+    for (const Item &i : cont) {
+      this->async_insert(i);
+    }
+    m_comm.barrier();
+  }
+
+  template <typename YGMContainer>
+  oversized_bag(ygm::comm          &comm,
+      const YGMContainer &yc) requires detail::HasForAll<YGMContainer> &&
+      detail::SingleItemTuple<typename YGMContainer::for_all_args>
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+
+    yc.for_all([this](const Item &value) { this->async_insert(value); });
+
+    m_comm.barrier();
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /**
+   * @todo we'll need to close all of the files we opened
+   */
+  ~oversized_bag() { m_comm.barrier(); }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  oversized_bag(const self_type &other)  // If I remove const it compiles
+      : m_comm(other.comm()), pthis(this), partitioner(other.comm()) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag(self_type &&other) noexcept
+      : m_comm(other.comm()),
+        pthis(this),
+        partitioner(other.comm()),
+        m_local_bag(std::move(other.m_local_bag)) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag &operator=(const self_type &other) { return *this = oversized_bag(other); }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /** @todo This might be hard to reimplement, we'll need to look at if we can swap on fstream object or if we'll have to swap data
+   * then have other.open() and this->open() to reopen the insert file
+   */
+  oversized_bag &operator=(self_type &&other) noexcept {
+    /* ---This is the original code---
+    std::swap(m_local_bag, other.m_local_bag);
+    */
+    return *this;
+  }
+
+  using detail::base_async_insert_value<oversized_bag<Item>, for_all_args>::async_insert;
+
+  void async_insert(const Item &value, int dest) {
+    auto inserter = [](auto pcont, const value_type &item) {
+      pcont->local_insert(item);
+    };
+
+    m_comm.async(dest, inserter, this->get_ygm_ptr(), value);
+  }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  void async_insert(const std::vector<Item> &values, int dest) {
+    auto inserter = [](auto pcont, const std::vector<Item> &values) {
+      for (const auto &v : values) {
+        pcont->local_insert(v);
+      }
+    };
+
+    m_comm.async(dest, inserter, this->get_ygm_ptr(), values);
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /** @todo */
+  void local_insert(const Item &val) { /*---This is the original code--- m_local_bag.push_back(val); */ }
```

Comment:
From here down to local_for_all, these would all be functions relating to file I/O, correct?

Reply:
All of the local_* functions will be handled by file i/o yes.